lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -728,6 +728,29 @@ void LLVMToSPIRVBase::transFPGAFunctionMetadata(SPIRVFunction *BF,
           new SPIRVDecorateFuseLoopsInFunctionINTEL(BF, Depth, Independent));
     }
   }
+  if (MDNode *ComponentII = F->getMetadata(kSPIR2MD::ComponentII)) {
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_invocation_pipelining_attributes)) {
+      if (size_t Cycles = getMDOperandAsInt(ComponentII, 0))
+        BF->addDecorate(new SPIRVDecorateInitiationIntervalINTEL(BF, Cycles));
+    }
+  }
+  if (MDNode *ComponentMaxConcurrency =
+          F->getMetadata(kSPIR2MD::ComponentMaxConcurrency)) {
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_invocation_pipelining_attributes)) {
+      size_t Invocations = getMDOperandAsInt(ComponentMaxConcurrency, 0);
+      BF->addDecorate(new SPIRVDecorateMaxConcurrencyINTEL(BF, Invocations));
+    }
+  }
+  if (MDNode *DisableComponentPipelining =
+          F->getMetadata(kSPIR2MD::DisableComponentPipelining)) {
+    if (BM->isAllowedToUseExtension(
+            ExtensionID::SPV_INTEL_invocation_pipelining_attributes)) {
+      if (size_t Enable = getMDOperandAsInt(DisableComponentPipelining, 0))
+        BF->addDecorate(new SPIRVDecoratePipelineEnableINTEL(BF, !Enable));
```

Comment:
It is a little bit confusing to read `Enable = get(Disable)`. Maybe to write it as:
```suggestion
      if (size_t Enable = (getMDOperandAsInt(DisableComponentPipelining, 0) == 0))
        BF->addDecorate(new SPIRVDecoratePipelineEnableINTEL(BF, Enable));
```

Reply:
Lets just call this variable 'disable'.
Problem with the code above is:
we shall add decoration in case of disable_pipelining metadata presents; this metadata carries '1' value =>
(getMDOperandAsInt(DisableComponentPipelining, 0) == 0) = false => we never enter this branch.